src/darwintracelib1.0/darwintrace.c

```diff
@@ -632,62 +683,144 @@ static inline bool __darwintrace_sandbox_check(const char *path, int flags) {
 
 	char command = -1;
 	char *t;
+    bool inserted_in_dtsm;
+
+    /*
+     * insertions to shared memory cache need not be always successful.
+     * It only allows most frequently occuring characters.
+     * Also, insertions are unsuccessful in rare cases and if that's not the case
+     * changes should be made.
+     * See LOWER_LIMIT, UPPER_LIMIT & POSSIBLE_CHARACTERS in dtsharedmemory.h
+     */
+
 
 	if (path[0] == '/' && path[1] == '\0') {
 		// Always allow access to /. Strange things start to happen if you deny this.
-		return true;
-	}
 
-	if ((flags & DT_ALLOWDIR) > 0) {
-		struct stat st;
-		if (-1 != lstat(path, &st) && S_ISDIR(st.st_mode)) {
-			return true;
-		}
+        //Store in shared memory cache
+        inserted_in_dtsm = __dtsharedmemory_insert(path, ALLOW_PATH);
+        if (!inserted_in_dtsm)
+            debug_printf("__dtsharedmemory_insert() failed for path %s\n", path);
+
+        return true;
+
 	}
+	 
 
 	// Iterate over the sandbox bounds and try to find a directive matching this path
 	for (__darwintrace_filemap_iterator_init(&filemap_it);
 	        (t = __darwintrace_filemap_iter(&command, &filemap_it));) {
 		if (__darwintrace_pathbeginswith(path, t)) {
 			switch (command) {
 				case FILEMAP_ALLOW:
-					return true;
+					
+                    //Store in shared memory cache
+                    inserted_in_dtsm = __dtsharedmemory_insert(t, ALLOW_PATH | IS_PREFIX);
+                    if (!inserted_in_dtsm)
+                        debug_printf("__dtsharedmemory_insert() failed for path prefix %s\n", t);
+                    
+                    return true;
+
 				case FILEMAP_ASK:
+
 					// ask the socket whether this file is OK
 					switch (dependency_check(path)) {
 						case 1:
-							return true;
+						    
+                            //Store in shared memory cache
+                            inserted_in_dtsm = __dtsharedmemory_insert(path, ALLOW_PATH);
+							if (!inserted_in_dtsm)
+                                debug_printf("__dtsharedmemory_insert() failed for path %s\n", path);
+                            
+                            return true;
+
 						case -1:
 							// if the file isn't known to MacPorts, allow
 							// access anyway, but report a sandbox violation.
 							// TODO find a better solution
 							if ((flags & DT_REPORT) > 0) {
 								__darwintrace_log_op("sandbox_unknown", path);
 							}
-							return true;
+
```

Comment:
@raimue , this is not the latest commit you are looking at. I had made one commit afterwards (in which I most probably resolved that problem by using flags instead of many bools). Would be great if you check that once.